lib/CMakeLists.txt

```diff
@@ -122,7 +120,16 @@ endif()
 # Clients of stp that don't use CMake will have to link the Boost libraries
 # in manually.
 # -----------------------------------------------------------------------------
-set(stp_link_libs ${MINISAT_LIBRARIES})
+# https://gitlab.kitware.com/cmake/cmake/-/issues/15415#note_633938
+
+file(COPY abc DESTINATION .)
+ADD_CUSTOM_COMMAND(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/abc/libabc.a
+    COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/abc && make libabc.a ABC_USE_PIC=ON -j
+)
```

Comment:
Has this been tested on macOS and Windows?

Reply:
Unfortunately, ABC's CMakefile is including the make as an implicit dependency to determine some options. If we hope to build ABC from the source, it's inevitable. 
https://github.com/berkeley-abc/abc/blob/db245f59bdf55041e3308c8753963f1366effedc/CMakeLists.txt#L62

Reply:
However, of course, if you really hope that, we can use `camke --build` or something like that to achieve the same result in `add_custom_command`. But I don't think the ABC will likely remove the make dependency someday because the official README is describing only the building instruction using `make`.  I think we should use `make` to avoid the possible issue by using the second-tier support building way.

Reply:
Interesting that they also call out to `make`.

Could you please share your thoughts on this part of my comment?

> I think this is the biggest obstacle to merging this. As ABC is also a CMake project it should work to include it within CMake's language, rather than calling out to another build tool. I read the described issue but it seems like there are other workarounds. And if ABC is not a private dependency, what's the issue—our library ends up with a handful of extra exported symbols? Doesn't seem like a big deal...

`ADD_CUSTOM_COMMAND` should be lowercase by the way. Please use the `WORKING_DIRECTORY` argument instead of the `cd` command. And yeah, using `cmake --build` does seem better if we absolutely have to call out this way.

Reply:
The reason we can't call `add_subdirectory`, is coming from the CMake issue I mentioned in the first comment. 
https://gitlab.kitware.com/cmake/cmake/-/issues/15415
If we do so, CMake will complain that the ABC will not be installed to the same destinations, and, as far as I know, this can be fixed only by using `$<BUILD_LOCAL_INTERFACE>`, CMake 3.26 feature. 

I strongly agree with that if we can use 'add_subdirectory()', it is better. However, I think you feel raising CMake version to 3.26 is not the fair deal for that. 
As a side note, I always prefer snap version of CMake for Ubuntu distirbutions because we can get the latest CMake by the canonical's official packaging system. Hence, if your concern is just to ease the installation on Ubuntu distributions, raising CMake to 3.26 or higher is not much problem, in my opinion. 

I guess that using 'cmake --build' seems to be the way to go although this will not remove the implicit dependency to 'make'.

Reply:
I agree with that using `WORKING_DIRECTORY` seems to be better.

Reply:
Fixed to use `cmake --build`.